inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp

```diff
@@ -432,52 +435,67 @@ void MKLDNNPlugin::MKLDNNInferRequest::changeDefaultPtr() {
     for (auto& it : externalPtr) {
         auto input = graph->inputNodesMap.find(it.first);
         if (input != graph->inputNodesMap.end()) {
-            if (input->second->getChildEdgeAt(0)->getMemory().GetPrimitive().get_data_handle() == it.second)
+            MKLDNNNodePtr inputNodePtr = input->second;
+            if (inputNodePtr->getChildEdgeAt(0)->getMemory().GetPrimitive().get_data_handle() == it.second)
                 continue;
+            auto childEdges = inputNodePtr->getChildEdges();
             // Input cannot be in-place with other primitives
             bool canBeInPlace = true;
-            for (size_t i = 0; canBeInPlace && i < input->second->getChildEdges().size(); i++) {
-                auto& child = input->second->getChildEdgeAt(i)->getChild();
-                if (child->isConstant())
-                    canBeInPlace = false;
+            for (auto& childEdge : childEdges) {
+                auto ce = childEdge.lock();
+                auto& child = ce->getChild();
 
+                if (child->isConstant()) {
+                    canBeInPlace = false;
+                    break;
+                }
                 auto* concat = dynamic_cast<MKLDNNConcatNode *>(child.get());
-                if (canBeInPlace && concat && concat->isOptimized())
+                if (concat && concat->isOptimized()) {
```

Comment:
It's incorrect because `concat` will be bool type inside condition so we cannot call `isOptimized()` for this variable. So I added type checking and `dynamic_cast` with call `isOptimized` inside condition